core/libdeno/api.cc

```diff
@@ -68,6 +68,8 @@ Deno* deno_new(deno_config config) {
       // If no snapshot is provided, we initialize the context with empty
       // main source code and source maps.
       deno::InitializeContext(isolate, context);
+    } else {
+      delete[] d->snapshot_.data;
```

Comment:
@ry adding this bit makes LSAN happy. I followed this [test](https://chromium.googlesource.com/v8/v8.git/+/4.5.103.9/test/cctest/test-serialize.cc#661) and it indeed seems to fix the issue.

Reply:
But makes tests fail 😓 still digging...

Reply:
Okay so I figured what causes this leak, however I'm not yet sure how to fix it.

The problem is in `will_snapshot` test case in `//cli/isolate.rs`; this test does a snapshot of a short script and then tries to instantiate a new isolate with that snapshot. `Isolate.snapshot()` returns `Snapshot1` which explicitly says in comment:
```
/// The type returned from deno_snapshot_new. Needs to be dropped.
```
During creation of new isolate (`Isolate::new`) this snapshot is passed in `deno_config` as `load_snapshot` (which has a type of `Snapshot2` not `Snapshot1`). However there is no way to tell that this is the type of snapshot on C++ side and Rust's borrow checker prevents an explicitly drop after calling `libdeno::deno_new`.